components/entitlement/org.wso2.carbon.identity.entitlement/src/test/java/org/wso2/carbon/identity/entitlement/persistence/HybridPolicyPersistenceManagerTest.java

```diff
@@ -0,0 +1,337 @@
+/*
+ * Copyright (c) 2024, WSO2 LLC. (http://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.entitlement.persistence;
+
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+import org.wso2.carbon.identity.common.testng.WithCarbonHome;
+import org.wso2.carbon.identity.common.testng.WithH2Database;
+import org.wso2.carbon.identity.common.testng.WithRealmService;
+import org.wso2.carbon.identity.common.testng.WithRegistry;
+import org.wso2.carbon.identity.entitlement.PDPConstants;
+import org.wso2.carbon.identity.entitlement.dto.PolicyDTO;
+import org.wso2.carbon.identity.entitlement.dto.PolicyStoreDTO;
+import org.wso2.carbon.identity.entitlement.internal.EntitlementConfigHolder;
+import org.wso2.carbon.identity.entitlement.internal.EntitlementServiceComponent;
+
+import java.util.List;
+import java.util.Properties;
+
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertFalse;
+import static org.testng.Assert.assertNull;
+import static org.testng.Assert.assertTrue;
+
+/**
+ * This class tests the behavior of the Hybrid Policy Persistence Manager class.
+ */
+@WithCarbonHome
+@WithRegistry(injectToSingletons = {EntitlementServiceComponent.class})
+@WithRealmService(injectToSingletons = {EntitlementConfigHolder.class}, initUserStoreManager = true)
+@WithH2Database(jndiName = "jdbc/WSO2IdentityDB", files = {"dbscripts/h2.sql"})
+public class HybridPolicyPersistenceManagerTest extends PolicyPersistenceManagerTest {
+
+    private JDBCPolicyPersistenceManager jdbcPolicyPersistenceManager;
+    private RegistryPolicyPersistenceManager registryPolicyPersistenceManager;
+
+    @BeforeMethod
+    public void setUp() throws Exception {
+
+        setPrivateStaticField(PersistenceManagerFactory.class, "POLICY_STORAGE_TYPE", "hybrid");
```

Comment:
Why we need this?

Reply:
This was added due to having incorrect POLICY_STORAGE_TYPE value due to a mocked static value added from a previous test class. This was removed and that static mocked value was reset back to the default value via the commit https://github.com/wso2/carbon-identity-framework/pull/5931/commits/e48e220b0688ef4f8f1c6867c2ca077a330347a4